File: app/src/main/java/com/dcrandroid/activities/EncryptWallet.java
package com.dcrandroid.activities;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.ActivityCompat;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.dcrandroid.MainApplication;
import com.dcrandroid.data.Constants;
import com.dcrandroid.util.DcrConstants;
import com.dcrandroid.MainActivity;
import com.dcrandroid.R;
import com.dcrandroid.util.PreferenceUtil;
import com.dcrandroid.util.Utils;

import mobilewallet.LibWallet;

/**
 * Created by collins on 12/26/17.
 */

public class EncryptWallet extends AppCompatActivity{

    private String seed;
    ProgressDialog pd;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_enter_passphrase);

        pd = Utils.getProgressDialog(EncryptWallet.this, false,false,"");

        final EditText passPhrase = findViewById(R.id.passphrase);
        final EditText verifyPassPhrase = findViewById(R.id.verifyPassphrase);
        Button encryptWallet = findViewById(R.id.button_encrypt_wallet);

        Intent i = getIntent();
        Bundle b = i.getExtras();

        if(b != null)
            seed = b.getString("seed");
            encryptWallet.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                final String pass = passPhrase.getText().toString();
                if(pass.equals("")){
                    Toast.makeText(EncryptWallet.this, R.string.enter_a_passphrase, Toast.LENGTH_SHORT).show();
                    return;
                }
                if(pass.equals(verifyPassPhrase.getText().toString())){
                    new Thread(){
                        public void run(){
                            try {
                                DcrConstants constants = DcrConstants.getInstance();
                                LibWallet wallet = constants.wallet;
                                if (wallet == null){
                                    return;
                                }
                                show("Creating wallet...");
                                wallet.createWallet(pass, seed);
                                runOnUiThread(new Runnable() {
                                    @Override
                                    public void run() {
                                        pd.dismiss();
                                        Intent i = new Intent(EncryptWallet.this, MainActivity.class);
                                        i.putExtra("passphrase", pass);
                                        startActivity(i);
                                        //Finish all the activities before this
                                        ActivityCompat.finishAffinity(EncryptWallet.this);
                                    }
                                });
                            } catch (final Exception e) {
                                e.printStackTrace();
                                runOnUiThread(new Runnable() {
                                    @Override
                                    public void run() {
                                        if(pd.isShowing()){
                                            pd.dismiss();
                                        }
                                        Toast.makeText(EncryptWallet.this, Utils.translateError(EncryptWallet.this, e), Toast.LENGTH_LONG).show();
                                    }
                                });
                            }
                        }
                    }.start();
                }else{
                    Toast.makeText(EncryptWallet.this, R.string.password_not_match,Toast.LENGTH_SHORT).show();
                }
            }
        });
    }

    public void show(final String str){
        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                pd.setMessage(str);
                pd.show();
            }
        });
    }
}

File: app/src/main/java/com/dcrandroid/activities/EncryptWallet.kt
/*
 * Copyright (c) 2018-2019 The Decred developers
 * Use of this source code is governed by an ISC
 * license that can be found in the LICENSE file.
 */

package com.dcrandroid.activities

import android.graphics.Color
import android.os.Bundle
import android.view.View

import com.dcrandroid.R
import com.dcrandroid.data.Constants
import com.dcrandroid.fragments.PasswordFragment
import com.dcrandroid.fragments.PinFragment
import kotlinx.android.synthetic.main.activity_enter_passphrase.*

class EncryptWallet : BaseActivity(), View.OnClickListener {

    private var seed: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_enter_passphrase)

        seed = intent.extras.getString(Constants.SEED)

        val passwordFragment = PasswordFragment()
        passwordFragment.seed = seed
        supportFragmentManager.beginTransaction().replace(R.id.container, passwordFragment)
                .commit()

        layout_password.setOnClickListener(this)
        layout_pin.setOnClickListener(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        seed = null
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.layout_password -> {

                password_bottom_border.visibility = View.GONE
                pin_bottom_border.visibility = View.VISIBLE

                layout_password.setBackgroundColor(Color.parseColor("#F3F5F6"))
                layout_pin.setBackgroundColor(Color.WHITE)
                val passwordFragment = PasswordFragment()
                passwordFragment.seed = seed
                supportFragmentManager.beginTransaction().replace(R.id.container, passwordFragment)
                        .commit()
            }
            R.id.layout_pin -> {

                password_bottom_border.visibility = View.VISIBLE
                pin_bottom_border.visibility = View.GONE

                layout_pin.setBackgroundColor(Color.parseColor("#F3F5F6"))
                layout_password.setBackgroundColor(Color.WHITE)
                val pinFragment = PinFragment()
                pinFragment.seed = seed
                supportFragmentManager.beginTransaction().replace(R.id.container, pinFragment)
                        .commit()
            }
        }
    }
}